corehq/apps/api/domain_metadata.py

```diff
@@ -62,7 +76,16 @@ def obj_get_list(self, bundle, **kwargs):
         if kwargs.get('domain'):
             return [self.obj_get(bundle, **kwargs)]
         else:
-            return list(Domain.get_all())
+            filters = {}
+            if hasattr(bundle.request, 'GET'):
+                filters = bundle.request.GET
+            domains = list(Domain.get_all())
```

Comment:
@lwyszomi I see here that you just copied it from above, but accessing all `Domain` documents like this in order to just get a few of them is very inefficient.

Looks like we're getting a sequence of requests like

```
GET /hq/admin/api/global/project_space_metadata/?limit=10&offset=0
GET /hq/admin/api/global/project_space_metadata/?limit=10&offset=10
GET /hq/admin/api/global/project_space_metadata/?limit=10&offset=20
GET /hq/admin/api/global/project_space_metadata/?limit=10&offset=30
...
```
—in total since Aug 29 we've gotten over 2000 request like this. If the goal is to get a complete list of domains, this method is O(N^2) instead of O(N) where N is the total number of domains.

@fyniac will be reaching out to you shortly, but can you come up with a way to do this such that the entire process of pulling all domains is O(N)?